packages/nx/src/tasks-runner/utils.ts

```diff
@@ -107,7 +107,40 @@ class InvalidOutputsError extends Error {
   }
 }
 
+class InvalidOutputsTypeError extends Error {
+  static maybeGetError(outputs: string[]): InvalidOutputsTypeError | undefined {
+    if (!Array.isArray(outputs)) {
+      return new InvalidOutputsTypeError(
+        "The 'outputs' field must be an array"
+      );
+    }
+
+    const typesArray = [];
+    let hasInvalidType = false;
+    for (const output of outputs) {
+      if (typeof output !== 'string') {
+        hasInvalidType = true;
+      }
+      typesArray.push(typeof output);
+    }
+
+    if (hasInvalidType) {
+      return new InvalidOutputsTypeError(
+        `The 'outputs' field must contain only strings, but received types: [${typesArray.join(
+          ', '
+        )}]`
+      );
+    }
+  }
+}
+
 export function validateOutputs(outputs: string[]) {
+  const invalidOutputsTypeError =
+    InvalidOutputsTypeError.maybeGetError(outputs);
+  if (invalidOutputsTypeError) {
+    throw invalidOutputsTypeError;
+  }
```

Comment:
Can you change this to an `assert` function please?

Something like `assertOutputsAreValidType`